pkg/controller/logstorage/logstorage_controller.go

```diff
@@ -348,9 +347,17 @@ func (r *ReconcileLogStorage) Reconcile(request reconcile.Request) (reconcile.Re
 
 		var flowShards = calculateFlowShards(ls.Spec.Nodes, defaultElasticsearchShards)
 		clusterConfig = render.NewElasticsearchClusterConfig(render.DefaultElasticsearchClusterName, ls.Replicas(), defaultElasticsearchShards, flowShards)
-		if err := r.client.Get(ctx, client.ObjectKey{Name: render.ElasticsearchStorageClass}, &storagev1.StorageClass{}); err != nil {
+
+		// Check if there is a StorageClass available to run Elasticsearch on.
+		scn := ls.Spec.StorageClassName
```

Comment:
This is a small thing, but since you're writing back to ls.Spec.StorageClassName, there's really no need for another variable. ls.Spec.StorageClassName will always have a non empty name.